testing/README.md

```diff
@@ -0,0 +1,390 @@
+# `import {Testing} from 'Shopify'`
+
+This guide looks at our recommended tools and conventions for testing JavaScript at Shopify. The primary focus of this guide is unit testing, but you can find some discussion of other types of testing in the [types of testing section](#types-of-testing).
+
+## Table of contents
+
+1. [Tools](#tools)
+1. [Style](#style)
+1. [Best practices](#best-practices)
+1. [Types of testing](#types-of-testing)
+1. [React](#react)
+1. [Resources](#resources)
+
+
+
+## Tools
+
+We recommend using the following tools in your tests:
+
+- [**mocha**](https://mochajs.org/) is our preferred test framework. It provides the basic scaffolding for tests including hooks, excellent support for asynchronous code, and more.
+
+- [**chai**](http://chaijs.com/) is our preferred assertion library. It provides many expressive assertions that have descriptive error messages in the case of test failures.
+
+- [**sinon**](http://sinonjs.org/) is our preferred library for handling spies and mocks in JavaScript.
+
+These tools are commonly used together and have many excellent online resources, including solid documentation. In addition to the above, you can use [chai plugins](http://chaijs.com/plugins/) to provide more meaningful assertions. Use caution, however: only include a plugin where it materially helps improve the legibility of tests (otherwise, the burden of other developers tracking down a particular assertion outweighs the benefits you get from having it). [chai-as-promised](http://chaijs.com/plugins/chai-as-promised/) is an excellent example of a chai plugin that makes otherwise awkward tests much easier to read.
+
+In addition to the testing framework, you generally need a test *runner*. Which runner to use will depend on the nature of your project:
+
+- For pure-JavaScript projects that do not need access to the DOM, avoid using a test runner and simply run the `mocha` CLI instead (if you need a simple DOM without actual rendering, you can also use [cheerio](https://github.com/cheeriojs/cheerio) to construct a fake DOM).
```

Comment:
IMO `JSDOM` is a better choice for testing code that interfaces with the browser as it exposes a browser-like API, not just a jQuery API subset for interfacing with a lightweight DOM.

For example, you can't do this with `cheerio`:

```js
import { jsdom } from 'jsdom';

const domDocument = jsdom('');
const myElement = domDocument.createElement('div');
myElement.setAttribute('data-my-element', '');

domDocument.body.appendChild(myElement);

const foundElement = domDocument.querySelector('[data-my-element]');

// true
console.log(foundElement === myElement);
```

It even says in the `cheerio` README

> Cheerio will not solve all your problems. I would still use JSDOM if I needed to work in a browser-like environment on the server, particularly if I wanted to automate functional tests.